project-management/src/main/java/life/qbic/projectmanagement/application/measurement/validation/MeasurementNGSValidator.java

```diff
@@ -226,6 +209,20 @@ ValidationResult validationProjectRelation(SampleCode sampleCode, ProjectId proj
           List.of("Sample ID does not belong to this project: %s".formatted(sampleCode.code())));
     }
 
+    @PreAuthorize("hasPermission(#projectId,'life.qbic.projectmanagement.domain.model.project.Project','READ')")
+    ValidationResult validationExperimentRelation(SampleCode sampleCode, String experimentId,
+        ProjectId projectId) {
+
+      boolean sampleContainedInExperiment = sampleInformationService.retrieveSamplesForExperiment(
+          projectId,
+          experimentId).stream().anyMatch(it -> it.sampleCode().equals(sampleCode));
+      if (sampleContainedInExperiment) {
+        return ValidationResult.successful();
+      }
+      return ValidationResult.withFailures(
+          List.of("Sample ID does not belong to this experiment: %s".formatted(sampleCode.code())));
```

Comment:
Keep the reason short to increase readability in the validation output. E.g.: Unknown Sample ID: Cannot match sample to experiment